src/containers/VideoContainer/VideoContainer.js

```diff
@@ -5,26 +5,66 @@ import { View } from 'react-native'
 import { Button, Text } from 'react-native-paper'
 import { Video } from 'expo-av'
 import PropTypes from 'prop-types'
+import { FILE_SERVER_BASE_URL } from '../../constants'
+import { STACK_SCREENS } from '../CTNavigationContainer/index'
 import styles from './VideoContainer.style'
 
 // CT requires this header as an addititonal security measure. Since we're not an approved referer, we can actually
 // just hardcode a valid referer and the API accepts it.
 const REFERER =
   'https://classtranscribe-dev.ncsa.illinois.edu/video?id=c79700ac-c3fc-439f-95c2-0511a1092862'
 
-const VideoContainer = ({ url }) => {
+const VideoContainer = ({ videos, index, navigation }) => {
+  // Passing a list of videos would not be a bad approach because react pass object as reference
+  const url = videos[index].video.video1Path
   const video = React.useRef(null)
+  const initPos = videos[index].watchHistory?.json?.timestamp || 0
+
   const [status, setStatus] = React.useState({
     isMuted: false,
     isPlaying: false,
     rate: 1.0,
   })
 
   const videoSource = {
-    uri: url,
+    uri: FILE_SERVER_BASE_URL + url,
     headers: { referer: REFERER },
   }
 
+  const renderLinkVideo = () => {
+    const prev = {
+      videos,
+      index: index - 1,
+    }
+    const next = {
+      videos,
+      index: index + 1,
+    }
+    return (
+      <View style={styles.buttons}>
+        <Button
+          mode="contained"
+          disabled={index === 0}
+          onPress={() => {
+            navigation.pop()
+            navigation.push(STACK_SCREENS.VIDEO, prev)
```

Comment:
I think it'd be better to store index as state within this component, then just change the state whenever these buttons are pressed. That way we don't have to push and pop a new screen every time the next video is requested -- it should help with performance a bit.